src/armory/CopyCasusCodePopup.js

```diff
@@ -0,0 +1,95 @@
+//@flow strict
+
+import * as React from 'react';
+import Popup from 'reactjs-popup';
+import Tank from '../tanks/Tank.js';
+import { ToastContainer , toast } from 'react-toastify';
+import { updateCasusCode } from '../globalComponents/apiCalls/tankAPIIntegration';
+
+type Props = {|
+	selectedTank: Tank,
+	usersTanks: Array<Tank>,
+|};
+
+type State = {|
+	popupOpen: boolean,
+	tankBeingCopiedFrom: ?Tank,
+|};
+
+class CopyCasusCodePopup extends React.Component<Props, State> {
+	
+	constructor(props: Props) {
+		super(props);
+
+		this.state = {
+			popupOpen: false,
+			tankBeingCopiedFrom : null,
+		}
+	}
+
+	setTankBeingCopiedFrom(tankBeingUsedId: string): void {
```

Comment:
I would double check to make sure you can't just make the value of these option tags just a tank object. I was under the impression that you could. That would make pretty much this entire method one line which would be very helpful.

Reply:
If you do the value becomes [Object object] and not the actual object